### ISSUE TITLE
Plugin incompatible with EDD 11.x

### ISSUE THREAD

**Issue by jjwfisher** [wulkanat/EDDX52#5]
Hi again!

I now have my x52 Pro - EDD does still work when loading DirectOutput.dll (slightly newer version from Logitech than what you provided for reference), and your EDDX52.dll.

However, I'm running into an issue when trying to run action files? I've included a screenshot for what happens when I try and run my test code (using the > button to the right of the program).

Also, the same error occurs when simply trying to continue, and save the now edited action file. I have used the Misc event class in EDD, but this error seems to persist no matter what I do.

I don't really understand the Action Lanaguage that much, nor can I find much helpful documentation on it. Do you think it would work better by using the version of EDD that you used when creating this extension? (I am currently running v 11.9.3, and I assume it has a newer version of the action language).

If this is unfixable/would require re-writing, please feel no pressure from me to do so - I feel like it would be a LOT of work!

![Screenshot_403](https://user-images.githubusercontent.com/62408547/107885663-8c024780-6ef3-11eb-93d0-69a0b7a043b2.png)
![Screenshot_404](https://user-images.githubusercontent.com/62408547/107885665-8c9ade00-6ef3-11eb-90e0-e4b93c88236c.png)

If you would like the full output code for either of these error messages, please say.

Thanks :)

**Comment by Theaninova** [comment]
Hmm, that looks to me more like an issue with EDDiscovery or an issue where the DLL crashing is not caught properly... The error, unfortunately, doesn't provide any real clues to what exactly went wrong and where.

**Comment by jjwfisher** [comment]
Just to update this, I have tried the same action program again with a fresh install of EDD v10.1.3, and it works flawlessly. I have created an issue on their github, and I will link it here if you want to see: [github issue](https://github.com/EDDiscovery/EDDiscovery/issues/2995)

Otherwise, since I use EDMC nominally, I will continue using the outdated version of EDD as it suits my needs, and enjoy programming the X52! Thank you for all your hard work creating this, hopefully I may be able to show what I've come up with if you're still interested in that :)

**Comment by Theaninova** [comment]
Thanks, I'm going to keep a watch on the EDD issue and hopefully be able to make changes if there was a breaking change I missed!

**Comment by Theaninova** [comment]
> hopefully I may be able to show what I've come up with if you're still interested in that :)

Absolutely, I never really came around to using this plugin so I'd love to include your setup as an example script, feel free to open an issue once you have something there!

**Comment by jjwfisher** [comment]
Sure!

I'd also like to add - no idea why - but I updated my EDD version to 11.9.3 following their reply on the issue I opened, and I'm no longer seeing the same issue at all. The DLL works pretty flawlessly so far, and I'm excited to do a sort of showcase when I'm done!

Only thing I haven't learnt yet is how to do MFD callbacks, going to sort them soon.

**Comment by Eahlstan** [comment]
Just for info:

https://github.com/EDDiscovery/EDDiscovery/issues/3318
